src/Datagrid/DatagridMapper.php

```diff
@@ -85,7 +85,7 @@ public function add(
                 array_merge($filterOptions, $fieldDescriptionOptions)
             );
         } else {
-            throw new \RuntimeException(
+            throw new \InvalidArgumentException(
```

Comment:
IMO, `\TypeError` seems better for this case.

Reply:
For the same reason, it should be something extending the LogicException for me